streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -395,7 +395,7 @@ public void pipeInput(final ConsumerRecord<byte[], byte[]> consumerRecord) {
                 consumerRecord.value())));
 
             // Process the record ...
-            ((InternalProcessorContext) task.context()).setRecordContext(new ProcessorRecordContext(consumerRecord.timestamp(), offset, topicPartition.partition(), topicName));
+            ((InternalProcessorContext) task.context()).setRecordContext(new ProcessorRecordContext(consumerRecord.timestamp(), offset, topicPartition.partition(), topicName, consumerRecord.headers()));
```

Comment:
Should we add a test do `TopologyTestDriverTest` that checks that headers are forwarded correctly?

What about `ConsumerRecordFactory` and `OutputVerifier`?  Can they be used to generate `ConsumerRecords` with headers and compare `ProducerRecords` with headers?

What about `MockProcessorContext` from `streams-test-utils` package?